db/memdb/iterator.go

```diff
@@ -0,0 +1,148 @@
+package memdb
+
+import (
+	"bytes"
+	"context"
+
+	tmdb "github.com/cosmos/cosmos-sdk/db"
+	"github.com/google/btree"
+)
+
+const (
+	// Size of the channel buffer between traversal goroutine and iterator. Using an unbuffered
+	// channel causes two context switches per item sent, while buffering allows more work per
+	// context switch. Tuned with benchmarks.
+	chBufferSize = 64
+)
+
+// memDBIterator is a memDB iterator.
+type memDBIterator struct {
+	ch     <-chan *item
+	cancel context.CancelFunc
+	item   *item
+	start  []byte
+	end    []byte
+}
+
+var _ tmdb.Iterator = (*memDBIterator)(nil)
+
+// newMemDBIterator creates a new memDBIterator.
```

Comment:
can you update this description? this is not explaining what this function does

Reply:
The function's not exported, so the inline comments are probably more useful, but I can add a summary

Reply:
Note: I didn't write this code, it's pulled verbatim from `tmdb`